packages/lit-node-client-nodejs/src/lib/lit-node-client-nodejs.ts

```diff
@@ -2858,6 +2881,70 @@ export class LitNodeClientNodeJs
     return signatures;
   };
 
+  /**
+   * Retrieves the PKP session signatures.
+   *
+   * @param params - The parameters for retrieving the PKP session signatures.
+   * @returns A promise that resolves to the PKP session signatures.
+   * @throws An error if any required parameter is missing or if both `litActionCode` and `ipfsId` are provided.
+   */
+  getPkpSessionSigs = async (
+    params: GetPkpSessionSigs
+  ): Promise<SessionSigsMap> => {
+    const chain = params?.chain || 'ethereum';
+
+    const pkpSessionSigs = this.getSessionSigs({
+      chain,
+      ...params,
+      authNeededCallback: async (props: AuthCallbackParams) => {
+        // -- validate
+        if (!props.expiration) {
+          throw new Error(
+            '[getPkpSessionSigs/callback] expiration is required'
+          );
+        }
+
+        if (!props.resources) {
+          throw new Error('[getPkpSessionSigs/callback]resources is required');
+        }
+
+        if (!props.resourceAbilityRequests) {
+          throw new Error(
+            '[getPkpSessionSigs/callback]resourceAbilityRequests is required'
+          );
+        }
+
+        // lit action code and ipfs id cannot exist at the same time
+        if (props.litActionCode && props.ipfsId) {
+          throw new Error(
+            '[getPkpSessionSigs/callback]litActionCode and ipfsId cannot exist at the same time'
+          );
+        }
```

Comment:
Shouldn't these checks be at the top of the function instead inside the callback?

Reply:
unrelated to this pr - yes you can do that, but this is for readability when user digging into the code, because this would be the assertions you have to do yourself if you were to pass in the callback yourself using `getSessionSigs`